src/main/java/com/ericsson/ei/controller/FailedNotificationControllerImpl.java

```diff
@@ -60,27 +64,38 @@
     public ResponseEntity<?> getFailedNotifications(
             @RequestParam(value = "subscriptionName", required = true) final String subscriptionName,
             final HttpServletRequest httpRequest) {
-        ObjectMapper mapper = new ObjectMapper();
-        QueryResponse queryResponse = new QueryResponse();
-        QueryResponseEntity queryResponseEntity = new QueryResponseEntity();
-        try {
-            List<String> response = processMissedNotification.processQueryMissedNotification(subscriptionName);
-            if (!response.isEmpty()) {
-                queryResponseEntity = mapper.readValue(response.get(0), QueryResponseEntity.class);
+        Set<String> subscriptionNameList = new HashSet<>(
+                Arrays.asList(subscriptionName.split(",")));
+        JSONArray foundArray = new JSONArray();
+        List<String> notFoundList = new ArrayList<>();
+
+        subscriptionNameList.forEach(name -> {
+            try {
+                List<String> response = processMissedNotification.processQueryFailedNotification(
```

Comment:
Can you not process all at once by dong MongoDB.find("subscription1 and Subscription2 etc...")?
You would still have to loop or reformat the query of more than one, but it would be just one call to MongoDB.

Reply:
This is how the code in subscription looks. It's just lifted.

Reply:
I looked into this and this change requires more work than I was comfortable with in this scope. Nothing is broken but the code could be improved in another story.